server/store/service.go

```diff
@@ -70,3 +65,19 @@ func (s *Service) hashkey(globalPrefix, namespace, prefix, id string) string {
 
 	return key
 }
+
+func hashkey(globalPrefix, namespace, prefix, id string) string {
+	namespacePrefixHash := make([]byte, 20)
+	sha3.ShakeSum128(namespacePrefixHash, []byte(namespace+prefix))
+
+	idHash := make([]byte, 16)
+	sha3.ShakeSum128(idHash, []byte(id))
+
+	encodedPrefix := make([]byte, ascii85.MaxEncodedLen(len(namespacePrefixHash)))
+	_ = ascii85.Encode(encodedPrefix, namespacePrefixHash)
+
+	encodedID := make([]byte, ascii85.MaxEncodedLen(len(idHash)))
+	_ = ascii85.Encode(encodedID, idHash)
+
+	return globalPrefix + path.Join(string(encodedPrefix), string(encodedID))
```

Comment:
Nit: why the `path.Join`? ASCII85 encodings contain slashes anyway, so you can't use them blindly to split keys to namespaces and IDs -- you have to rely on the known length of the namespace. Seems to me like the extra slash here is just taking up space.

Reply:
thx! missed it. was a leftover from the md5 code.